planner/mapper/mapper.go

```diff
@@ -738,25 +733,38 @@ func getTopLevelInfo(
 	store client.Store,
 	selectRequest *request.Select,
 	collectionName string,
-) (*core.DocumentMapping, client.Collection, error) {
+) (*core.DocumentMapping, client.SchemaDescription, error) {
 	mapping := core.NewDocumentMapping()
 
 	if _, isAggregate := request.Aggregates[selectRequest.Name]; isAggregate {
 		// If this is a (top-level) aggregate, then it will have no collection
 		// description, and no top-level fields, so we return an empty mapping only
-		return mapping, nil, nil
+		return mapping, client.SchemaDescription{}, nil
 	}
 
 	if selectRequest.Root == request.ObjectSelection {
-		mapping.Add(core.DocKeyFieldIndex, request.KeyFieldName)
-
+		var schema client.SchemaDescription
 		collection, err := store.GetCollectionByName(ctx, collectionName)
 		if err != nil {
-			return nil, nil, err
+			// If the collection is not found, check to see if a schema of that name exists,
+			// if so, this must be an embedded object
+			schemas, err := store.GetSchemasByName(ctx, collectionName)
+			if err != nil {
+				return nil, client.SchemaDescription{}, err
+			}
+			if len(schemas) == 0 {
+				return nil, client.SchemaDescription{}, NewErrTypeNotFound(collectionName)
+			}
+			// `schemas` will contain all versions of that name, as views cannot be updated atm this should
+			// be fine for now
+			schema = schemas[0]
```

Comment:
question: Should we try to ensure it's the latest version of the schema?

Reply:
I thought about that, but it felt like too much code right now.  If it was really cheap to do so I would probably have added it in, but it is ultimately dead-code atm given that there can only be one version. 

Better to wait until we actually implement something that makes this necessary (and testable), and then code it properly.